clang-tools-extra/clang-doc/Serialize.cpp

```diff
@@ -947,6 +952,53 @@ emitInfo(const NamespaceDecl *D, const FullComment *FC, Location Loc,
   return {std::move(NSI), makeAndInsertIntoParent<const NamespaceInfo &>(*NSI)};
 }
 
+static void parseFriends(RecordInfo &RI, const CXXRecordDecl *D) {
+  if (D->hasDefinition() && D->hasFriends())
```

Comment:
I think if you invert the condition you can use early return to avoid a level of nesting.